init/setup-kdump.functions

```diff
@@ -715,6 +715,44 @@ function kdump_get_targets()						   # {{{
     return 0
 }									   # }}}
 
+#
+# Get root device from the kernel command line.
+#
+# Output:
+#   string that contains the device name
+function kdump_get_root_from_commandline()
+{
+    local _o _val
+    unset _val
+    unset _o
+    CMDLINE=$(cat /proc/cmdline)
+    _arg="root"
+
+    for _o in $CMDLINE; do
```

Comment:
This for loop is naïve. The Linux kernel provides two ways to quote spaces inside arguments:

1. `"param=value with spaces"`
2. `param="value with spaces"`

This can bite you if someone passes a command line like this:
```
unrelated="value with root=bogus"
```
Which gets parsed as:

- `unrelated="value`
- `with`
- `root=bogus"`

Additionally, the kernel parser stops at `--` (after removing quotes).